autosklearn/estimators.py

```diff
@@ -951,10 +983,15 @@ def has_key(rv, key):
                     "start_time": pd.NA,
                     "end_time": pd.NA,
                     "status": pd.NA,
-                    "cost": pd.NA,
                     "train_loss": pd.NA,
                     "config_origin": pd.NA,
                 }
+                if num_metrics == 1:
```

Comment:
We could also always treat it as a list to avoid this if statement

Reply:
I disagree. If we always treat single-objective as a list, then the name of the cost for failing runs (here) will be `cost_{i}` while for succeeding runs it will only be `cost`.